File: dotfiles_manager/src/commands/activate.rs
use crate::config::command::{get_aliases, get_functions};

use super::structs::{Command, ToCommand};

pub fn display_activate_aliases_and_functions() {
    let aliases = get_aliases()
        .iter()
        .map(|declare| declare.to_command())
        .collect::<Vec<Command>>();

    let functions = get_functions()
        .iter()
        .map(|declare| declare.to_command())
        .collect::<Vec<Command>>();

    for command in aliases {
        println!("{}", command.definition);
    }

    for command in functions {
        println!("{}", command.definition);
    }
}

File: dotfiles_manager_v2/src/commands/activate.ts
import { getAliases, getFunctions } from '../config/commands.js';
import { toCommandFromAlias, toCommandFromFunction } from './types.js';

export function displayActivateAliasesAndFunctions(): string {
  const aliases = getAliases().map(toCommandFromAlias);
  const functions = getFunctions().map(toCommandFromFunction);
  
  const output: string[] = [];
  
  for (const command of aliases) {
    output.push(command.definition);
  }
  
  for (const command of functions) {
    output.push(command.definition);
  }
  
  return output.join('\n');
} 